service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -80,59 +79,122 @@ public String toString() {
+            //Get all values with a batch get
+            this.decisionKey = evalRequestKey.toDecisionKey();
+            this.entryKeys =  prepareKeys(evalRequestKey);
+            this.entryValues = multiGet(this.entryKeys);
+            this.lastValueIndex = this.entryValues.size() - 1;
+
+            logCacheGetDebugMessages(evalRequestKey, this.decisionKey, this.entryKeys, this.entryValues);
+
+            //create separate list of policySetTimes to prevent mutation on entryValues
+            for (int i = 0; i < evalRequestKey.getPolicySetIds().size(); i++) {
+                this.policySetTimestamps.add(this.entryValues.get(i));
+            }
+        }
+        
+        //Prepare keys to fetch in one batch            
+        private List<String> prepareKeys(final PolicyEvaluationRequestCacheKey evalRequestKey) {
+            List<String> keys = new ArrayList<>();
+
+            //Add 'n' Policy Set keys
+            LinkedHashSet<String> policySetIds = evalRequestKey.getPolicySetIds();
+            policySetIds.forEach(policySetId -> keys.add(policySetKey(evalRequestKey.getZoneId(), policySetId)));
+            
+            // n+1
+            keys.add(subjectKey(evalRequestKey.getZoneId(), evalRequestKey.getSubjectId()));
+
+            //n+2
+            keys.add(resourceKey(evalRequestKey.getZoneId(), evalRequestKey.getResourceId()));
+
+            //n+3
+            keys.add(this.decisionKey);
+            
+            return keys;
         }
 
-        if (isCachedRequestInvalid(values, new DateTime(cachedResult.getTimestamp()))) {
-            delete(keys.get(keys.size() - 1));
-            LOGGER.debug("Cached decision for key '{}' is not valid.", redisKey);
-            return null;
+        /**
+         * (eval result, eval time, resolved resource uri(s)). 
+         */
+        String getDecisionString() {
+            return entryValues.get(this.lastValueIndex);
         }
-        return cachedResult;
-    }
 
-    private List<String> assembleKeys(final PolicyEvaluationRequestCacheKey key) {
-        // Only need to assemble keys for policy sets, the subject, and the policy evaluation result.
-        // Resource-related information will be captured in the resolved resource URIs from the cached
-        // policy evaluation result.
-        List<String> keys = new ArrayList<>();
-        LinkedHashSet<String> policySetIds = key.getPolicySetIds();
-        policySetIds.forEach(policySetId -> keys.add(policySetKey(key.getZoneId(), policySetId)));
-        keys.add(subjectKey(key.getZoneId(), key.getSubjectId()));
-        keys.add(resourceKey(key.getZoneId(), key.getResourceId()));
-        keys.add(key.toRedisKey());
-        return keys;
-    }
+        String getDecisionKey() {
+            return decisionKey;
+        }
+
+        String getSubjectLastModified() {
+            return entryValues.get(lastValueIndex - 2);
+        }
+
+        String getRequestedResourceLastModified() {
```

Comment:
Changing the name to just `getResourceLastModified` should suffice (and is also consistent with the naming convention used for the other `get*LastModified` methods).